Update Users by ID Reference Topic (#52)
UWC2-APIDOC/to-do-service-public: docs/api/users-update-by-id.md

---
layout: page
---

# Update user by ID

Allows the editing of [`user`](user) objects that already exist. You will need to know the `id` parameter of the user you want to edit.

> [!IMPORTANT]
> The information for the user is replaced **entirely** by the request body, so if you want to keep information the same you still need to include it in the request body.

## URL

```shell
{server_url}/users/{id}
```

> 

Reviewer: What HTTP method do you use for this call?

Author: Added the method!

## Params

| Parameter name | Type | Description |
| -------------- | ------ | ------------ |
| `id` | number | The record ID of the user to return |

## Request headers

| Key | Value |
|---|---|
| Content-Type | application/json |

## Request body parameters

| Property name | Type | Description |
| ------------- | ----------- | ----------- |
| `last_name` | string | The user's last name |
| `first_name` | string | The user's first name |
| `email` | string | The user's email address |
| `id` | number | The user's unique record ID |

> 

Reviewer: Can you change the `id` value?

Can you just specify the property or properties that you want to change and leave the rest of them unchanged by not including them?

Author: You cannot change the ID, which I have noted in the parameters section.

I have an alert on lines 9-10 that addresses the need to include properties even if you want them unchanged. With this newest commit I have added a qualifier that ID remains the same regardless.

Reviewer: See my other comments above.

---
layout: page
---

# Update user by ID

Allows the editing of [`user`](user) objects that already exist. You will need to know the `id` parameter of the user you want to edit.

> > [!IMPORTANT]

Reviewer: This is a good (and important) note, but it might not be in the best location.

Because this has to do with the request body, maybe put it down in that section. Remember that reference topics are rarely read sequentially.

Author: Done!

> > The information for the user is replaced **entirely** by the request body, so if you want to keep information the same you still need to include it in the request body. The only exception to this is the ID which remains the same regardless of the body that is submitted.

Reviewer: ```suggestion
> The request body replaces the current properties of the `user`.
> Properties that you don't want to change, must have their current values assigned in the request body,
```

Something to try, if you like it. I've shortened it and put into active voice.

## HTTP Method

> PUT

Reviewer: I like it!
I just realized that none of the other reference topics mention this and they probably should. (That's not your problem, just something this brought to my attention.)

## URL

```shell
{server_url}/users/{id}
```

## Params

| Parameter name | Type | Description |
| -------------- | ------ | ------------ |
| `id` | number | The record ID of the user to return |

## Request headers

| Key | Value |
|---|---|
| Content-Type | application/json |

## Request body parameters

> 

Reviewer: Maybe add a note that describes that all properties (except for the `id`) are required and describe what happens if a property is omitted. (per comment above)

| Property name | Type | Description |
| ------------- | ----------- | ----------- |
| `last_name` | string | The user's last name |
| `first_name` | string | The user's first name |
| `email` | string | The user's email address |

> | `id` | number | This is not required and cannot be changed by this process |

Reviewer: I'm a little confused.
If this isn't required and it isn't used by the operation, why not just omit `id` from the request body?
Having the same property in two places (URL and request body) is confusing, especially when only one (the URL) is used.

Author: Good note. I'll remove it.

## Example Request Body

```json
{
    "last_name": "Bailey",
    "first_name": "Bill",

>     "email": "b.bailey@example.com",

Reviewer: ```suggestion
    "email": "b.bailey@example.com"
```

>     "id": "4"

Reviewer: ```suggestion
```
